hundred_x/client.py

```diff
@@ -30,23 +31,56 @@
 
 
 class HundredXClient:
+    private_functions: List[str] = [
+        "/v1/withdraw",
+        "/v1/order",
+        "/v1/order/cancel-and-replace",
+        "/v1/order",
+        "/v1/openOrders",
+        "/v1/orders",
+        "/v1/balances",
+        "/v1/positionRisk",
+        "/v1/approved-signers",
+        "/v1/session/login",
+        "/v1/referral/add-referee",
+        "/v1/session/logout",
+    ]
+    public_functions: List[str] = [
+        "/v1/products",
+        "/v1/products/{product_symbol}",
+        "/v1/trade-history",
+        "/v1/time",
+        "/v1/uiKlines",
+        "/v1/ticker/24hr",
+        "/v1/depth",
+    ]
+
     def __init__(
         self,
         env: Environment = Environment.TESTNET,
         private_key: str = None,
-        subaccount_id: int = 1,
+        subaccount_id: int = 0,
     ):
         """
         Initialize the client with the given environment.
         """
         self.env = env
         self.rest_url = APIS[env][ApiType.REST]
         self.websocket_url = APIS[env][ApiType.WEBSOCKET]
-        self.wallet = eth_account.Account.from_key(private_key)
-        self.public_key = self.wallet.address
-        self.subaccount_id = subaccount_id
+        if any([not self.rest_url, not self.websocket_url]):
+            raise UserInputValidationError(
+                f"Invalid environment: {env} Missing REST or WEBSOCKET URL for the environment."
+            )
+        if private_key:
```

Comment:
do you want a `set_private_key (private_key: str)` function to allow a user to enable private functions without having to instantiate a new client object? would just include these two lines

Reply:
Just having a default of none means as soon as there's anything in that pk var it will hit this logic block

Reply:
i just picked that line to show that there's currently only one way to set the private key, at init time

would be nice if it could be added after init, so we can add it without having to re-instantiate? i guess a minor benefit

instead of `client = client(private_key,...)` to access private functions you could do `client.set_private_key(private_key)`

Reply:
Yeah i see your point just want to get this merged and released
:)